orca-echo/src/main/groovy/com/netflix/spinnaker/orca/echo/pipeline/ManualJudgmentStage.groovy

```diff
@@ -73,20 +81,53 @@ class ManualJudgmentStage implements StageDefinitionBuilder, AuthenticatedStage
     final long timeout = TimeUnit.DAYS.toMillis(3)
 
     @Autowired(required = false)
-    EchoService echoService
+    private final EchoService echoService
+
+    @Autowired(required = false)
+    private final FiatPermissionEvaluator fiatPermissionEvaluator
+
+    @Autowired(required = false)
+    private FiatStatus fiatStatus
+
+    @Autowired(required = false)
+    private ManualJudgmentAuthzGroupsUtil mnlJudgUtil
+
+    @Autowired(required = false)
+    private ObjectMapper objectMapper
+
+    WaitForManualJudgmentTask(Optional<EchoService> echoService, Optional<FiatPermissionEvaluator> fpe,
+                              Optional<FiatStatus> fiatStatus, Optional<ObjectMapper> objectMapper,
+                              Optional<ManualJudgmentAuthzGroupsUtil> mnlJudgUtil) {
+      this.echoService = echoService.orElse(null)
+      this.fiatPermissionEvaluator = fpe.orElse(null)
+      this.fiatStatus = fiatStatus.orElse(null)
+      this.objectMapper = objectMapper.orElse(null)
+      this.mnlJudgUtil = mnlJudgUtil.orElse(null)
+    }
 
     @Override
     TaskResult execute(StageExecution stage) {
       StageData stageData = stage.mapTo(StageData)
+      def username = AuthenticatedRequest.getSpinnakerUser().orElse(stage.lastModified ? stage.lastModified.user : "")
+      boolean fiatEnabled = fiatStatus ? fiatStatus.isEnabled() : false
+      boolean isAuthorized = false
+      def appPermissions
+      def stageRoles
+      if (fiatEnabled) {
+        appPermissions = getApplicationPermissions(stage)
```

Comment:
I'd rather we not introduce a call to `front50` if there aren't any required roles here. The majority of use-cases (and certainly everything pre-existing) will not need this.

Reply:
done..